app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -224,76 +280,189 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status) {
                 BluetoothGattService gattService = mBluetoothGatt.getService(xDripDataService);
                 if (gattService != null) {
                     BluetoothGattCharacteristic gattCharacteristic = gattService.getCharacteristic(xDripDataCharacteristic);
-                    if (gattCharacteristic != null ) {
+                    if (gattCharacteristic != null) {
                         final int charaProp = gattCharacteristic.getProperties();
-
                         if ((charaProp | BluetoothGattCharacteristic.PROPERTY_NOTIFY) > 0) {
-                            mBluetoothGatt.setCharacteristicNotification(gattCharacteristic, true);
+                            if ((charaProp | BluetoothGattCharacteristic.PROPERTY_NOTIFY) > 0) {
+                                mBluetoothGatt.setCharacteristicNotification(gattCharacteristic, true);
+                            } else {
+                                Log.e(TAG, "onServicesDiscovered: characteristic " + gattCharacteristic.getUuid() + " doesn't have notify properties");
+                            }
                         } else {
-                            Log.e(TAG, "characteristic " + gattCharacteristic.getUuid() + " doesn't have notify properties");
+                            Log.e(TAG, "onServicesDiscovered: characteristic " + xDripDataCharacteristic + " not found");
                         }
                     } else {
-                        Log.e(TAG, "characteristic " + xDripDataCharacteristic + " not found");
+                        Log.e(TAG, "onServicesDiscovered: service " + xDripDataService + " not found");
+                        //Log.v(TAG, "onServicesDiscovered: Already have that service");
                     }
-                } else {
-                    Log.e(TAG, "service " + xDripDataCharacteristic + " not found");
+                    Log.w(TAG, "onServicesDiscovered received success: " + status);
                 }
+            } else {
+                Log.w(TAG, "onServicesDiscovered received: " + status);
             }
         }
 
         @Override
-        public void onCharacteristicChanged(BluetoothGatt gatt, BluetoothGattCharacteristic characteristic) {
-            final byte[] data = characteristic.getValue();
-            if (data != null && data.length > 0) { setSerialDataToTransmitterRawData(data, data.length); }
+        public void onCharacteristicRead(BluetoothGatt gatt,
```

Comment:
All fixed Tzachi. It was to do with the way I was initiating PebbleSync. I figured out the error of my ways and have corrected it. Many thanks for the follow up. 

 
John Stevens
M - +61 418526349

> On 20 May 2015, at 07:08, tzachi-dar <notifications@github.com> wrote:
> 
> In app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java:
> 
> >              }
> >          }
> >  
> >          @Override
> > -        public void onCharacteristicChanged(BluetoothGatt gatt, BluetoothGattCharacteristic characteristic) {
> > -            final byte[] data = characteristic.getValue();
> > -            if (data != null && data.length > 0) { setSerialDataToTransmitterRawData(data, data.length); }
> > +        public void onCharacteristicRead(BluetoothGatt gatt,
> Hi John, Seems like you had some question about BgSendQueue. Have you been able to solve it or can you please ask again? Thanks Tzachi
> …
> —
> Reply to this email directly or view it on GitHub.
>